src/components/DocumentsTable.tsx

```diff
@@ -107,6 +121,39 @@ export function DocumentsTable(props: Props) {
       });
   };
 
+  useEffect(() => {
+    if (reactSettings.show_document_preview) {
+      props.documents.forEach(async ({ id, hash, type }) => {
+        const alreadyExist = documentsUrls.find(
+          ({ documentId }) => documentId === id
+        );
+
+        if (!alreadyExist && (type === 'png' || type === 'jpg')) {
+          await queryClient.fetchQuery(
```

Comment:
Looks like we're missing caching here. Can we cache infinitely?

['documents', hash]